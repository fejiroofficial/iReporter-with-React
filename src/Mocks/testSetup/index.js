/* eslint-disable import/no-extraneous-dependencies */
import '@babel/polyfill';
import React from 'react';
import { Provider } from 'react-redux';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import configMockStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';


export const mockState = {
  user: {
    token: '',
    isLoggedIn: false,
    loginStatus: false,
    message: '',
    isLoading: false,
    profile: {
      firstname: 'John',
      username: 'johnny',
      image: 'image.jpg',
      errorData: [],
    },
    resetPassword: {
      loading: false,
      message: '',
    }
  },
  bookmark: {
    loading: false,
    articles: [],
    message: '',
  },
  article: {
    newComments: {
      body: 'Hello world',
      id: 1,
      authorName: 'John',
      authorImage: 'image.jpg',
      date: 'July 4th 2018',
    },
    postingComment: false,
    commentMessage: 'Done',
    gettingComments: false,
    oldComments: [],
    hasMoreComments: false,
    commentPage: {
      current: 1,
      currentCount: 1,
      totalCount: 1,
    },
  },
  createArticle: {
    loading: false,
    message: false,
  },
  articlesReducer: {
    articles: []
  },
  articlesCategoryReducer: []
};


export const createMockStore = () => {
  const mockStore = configMockStore([thunk]);
  configure({ adapter: new Adapter() });
  return mockStore;
};

const setup = (component, initailState = mockState) => {
  const mockStore = configMockStore([thunk]);
  configure({ adapter: new Adapter() });
  const connectedWrapper = mount(
    <Provider store={mockStore(initailState)}>
      <BrowserRouter>
        {component}
      </BrowserRouter>
    </Provider>
  );
  return connectedWrapper;
};

export default setup;
