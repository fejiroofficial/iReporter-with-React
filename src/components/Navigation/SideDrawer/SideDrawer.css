.Overlay {    
  height: 100%;
  width: 100%;
  position: fixed;
  z-index: 1;
  left: 0;
  top: 0;
  background-color: rgb(32, 178, 170);
  background-color: rgba(32, 178, 170, 0.93);
  overflow-x: hidden; 
  transition: 0.5s; 
}

.OverlayContent {
  position: relative;
  top: 25%;
  width: 100%;
  text-align: center;
  margin-top: 10px;
}

.Overlay a {
  padding: 8px;
  text-decoration: none;
  font-size: 20px;
  color: ghostwhite;
  display: block;
  transition: 0.3s;
}

.Overlay a:hover, .overlay a:focus {
  color: #f1f1f1;
}

.Overlay .Closebtn {
  position: absolute;
  top: 20px;
  right: 1px;
  font-size: 45px;
}

.Seperator {
  height: 1px;
  width: inherit;
  background-color: lightgrey;
}
