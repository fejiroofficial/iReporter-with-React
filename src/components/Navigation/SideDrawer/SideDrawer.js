import React from 'react';
import { NavLink } from 'react-router-dom';
import Classes from './SideDrawer.css';

const sideDrawer = (props) => (
  <div className={Classes.Overlay}>
    <button onClick={props.clicked} className={Classes.Closebtn}>&times;</button>
    <div className={Classes.OverlayContent}>
      <div className={Classes.Seperator} />
      <NavLink to="/about" >ABOUT US</NavLink>
      <div className={Classes.Seperator} />
      <NavLink to="/corruption" >LEARN ABOUT CORRUPTION</NavLink>
      <div className={Classes.Seperator} />
      <NavLink to="/redflags" >REDFLAGS</NavLink>
      <div className={Classes.Seperator} />
      <NavLink to="/interventions" >INTERVENTIONS</NavLink>
      <div className={Classes.Seperator} />
      <NavLink to="/login" >LOGIN</NavLink>
      <div className={Classes.Seperator} />
      <NavLink to="/signup" >SIGN UP</NavLink>
      <div className={Classes.Seperator} />
    </div >
  </div >
);

export default sideDrawer;
