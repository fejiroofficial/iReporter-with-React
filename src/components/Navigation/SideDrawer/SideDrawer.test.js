import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { createMockStore } from '../../../Mocks/testSetup';
import SideDrawer from './SideDrawer';

createMockStore();

describe('<NavigationItems>', () => {
  it('should render without crashing', () => {
    const container = (
      <SideDrawer>
        <p>Hello</p>
      </SideDrawer>
    );
    const wrapper = shallow(container);
    expect(wrapper).toMatchSnapshot();
  });
});
