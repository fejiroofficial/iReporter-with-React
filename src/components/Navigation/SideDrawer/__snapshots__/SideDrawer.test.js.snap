// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItems> should render without crashing 1`] = `
<div>
  <button>
    Ã—
  </button>
  <div>
    <div />
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/about"
    >
      ABOUT US
    </NavLink>
    <div />
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/corruption"
    >
      LEARN ABOUT CORRUPTION
    </NavLink>
    <div />
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/redflags"
    >
      REDFLAGS
    </NavLink>
    <div />
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/interventions"
    >
      INTERVENTIONS
    </NavLink>
    <div />
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/login"
    >
      LOGIN
    </NavLink>
    <div />
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/signup"
    >
      SIGN UP
    </NavLink>
    <div />
  </div>
</div>
`;
